plugin/mlsdbonlinelocator.cpp

```diff
@@ -100,14 +101,27 @@ bool MlsdbOnlineLocator::findLocation(const QList<MlsdbProvider::CellPositioning
         return false;
     }
 
-    QNetworkRequest req(QUrl("https://location.services.mozilla.com/v1/geolocate?key=" + m_mlsKey));
-    req.setHeader(QNetworkRequest::ContentTypeHeader, "application/json");
-
     QVariantMap map;
-    map.unite(globalFields());
-    map.unite(cellTowerFields(cells));
-    map.unite(wifiAccessPointFields());
+
+    QVariantMap map_cellTowerFields = cellTowerFields(cells);
+    if (!map_cellTowerFields.isEmpty()) map.unite(map_cellTowerFields);
```

Comment:
map.unite(empty_map) should be no-op anyway? This is not performance critical code so simplicity should win.

Reply:
This goal: 1) do not send empty requests to the service 2) send a request only when there is information on the cellular and wifi networks. With this code, we reduce the number of requests and reduce the load on the wireless modem

Reply:
I don't you answered the question part here. If cellTowerFields() returns an empty map, both old and new and code in this function should end up exactly the same, no?

Reply:
QVariantMap MlsdbOnlineLocator::cellTowerFields
{
    ...
    map["cellTowers"] = cellTowers;
    return map;
}
QVariantMap MlsdbOnlineLocator::wifiAccessPointFields()
{
    ...
    map["wifiAccessPoints"] = wifiInfoList;
    return map;
} 

Map not empty

Reply:
QVariantMap MlsdbOnlineLocator::cellTowerFields
{
    ...
    map["cellTowers"] = cellTowers;
    return map;
}
QVariantMap MlsdbOnlineLocator::wifiAccessPointFields()
{
    ...
    map["wifiAccessPoints"] = wifiInfoList;
    return map;
} 

Map not empty

Reply:
Meant old code in this function: map.unite(cellTowerFields(cells));

Reply:
why is that not good enough?
https://github.com/mer-hybris/geoclue-providers-mlsdb/commit/77ad5bff1135c6fde6facc7a42418f43cd660232#diff-b1cb97109ad45f6f0ecced027e7a540fR97
and
https://github.com/mer-hybris/geoclue-providers-mlsdb/commit/77ad5bff1135c6fde6facc7a42418f43cd660232#diff-b1cb97109ad45f6f0ecced027e7a540fR80

Reply:
I got the real data map from old code: 
QMap(("carrier", QVariant(QString, ""))("cellTowers", QVariant(QVariantList, ()))("considerIp", QVariant(bool, true))("fallbacks", QVariant(QVariantMap, QMap(("ipf", QVariant(bool, false))("lacf", QVariant(bool, false)))))("homeMobileCountryCode", QVariant(QString, ""))("homeMobileNetworkCode", QVariant(QString, ""))("wifiAccessPoints", QVariant(QVariantList, ())))
The status check "if (map.isEmpty())" is useless. In the old code, the function always sends a request to the server. In any case, the map has data